### DISCUSSION TITLE
Spectacle not compatible with windows version

### DISCUSSION THREAD

**Discussion by AndyK28** [getgauge/gauge#2654] [General]
```bash
PS D:\directory> gauge docs spectacle specs/
Error occurred while waiting for plugin process to finish.
Error : exec: not started
Error ----------------------------------

[Gauge]
Spectacle 0.2.0. fork/exec .\spectacle.exe: This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.       

Get Support ----------------------------
        Docs:          https://docs.gauge.org
        Bugs:          https://github.com/getgauge/gauge/issues
        Chat:          https://github.com/getgauge/gauge/discussions

Your Environment Information -----------
        windows, 1.6.10, 5884aa1
        dotnet (0.7.2), html-report (4.3.1), screenshot (0.3.0), spectacle (0.2.0)```

**Comment by chadlwilson** [comment]
Which version of windows and.which hardware are you running?

**Comment by mschreiber16** [comment]
I have confirmed this on Windows 10 Enterprise Version: 21H2.

**Reply by chadlwilson** [reply]
Did an older plugin version work ok e.g 0.1.4?

That's an EOL version of windows so I'm not personally going to spend much time trying to fix anything if it's specific to that version and can't be replicated on a modern Windows 10 version.

**Reply by chadlwilson** [reply]
Likely the problem is either something to do with x86 versions being removed (if you are not running Windows 10 in 64-bit mode) or something to do with modern golang versions not cross-compiling correctly for that Windows version without some special flags.

**Reply by mschreiber16** [reply]
I am running in 64 bit and another colleague is running Windows 11 Enterprise 23H2 and experiences the same error.

**Reply by chadlwilson** [reply]
if you can try `0.1.4` and let me know if that works, that'd help.

**Reply by mschreiber16** [reply]
> if you can try `0.1.4` and let me know if that works, that'd help.

That worked on Windows 11.  Is that something I can try on Windows 10?

**Reply by chadlwilson** [reply]
Yeah should be.

At least that helps confirm that something regressed (likely with the golang upgrade) in the 0.2.0 version of the plugin 😓

**Reply by chadlwilson** [reply]
Unfortunately I still can't replicate this making it difficult to figure out where/how to fix. Which approach did you use to install Gauge itself?

**Reply by chadlwilson** [reply]
Do you happen to use/develop for golang yourselves on these hosts? Do you have `GOARCH` or `GOHOSTARCH` set in environment or anything like that on these machines?

I can only find things like https://github.com/golang/go/issues/64530

I've tried rebuilding the plugin in https://github.com/getgauge/spectacle/releases/tag/v0.2.1 in a desperate attempt to see if that does anything to fix the issue. Can you let me know if `0.2.1` fixes the problem?

**Reply by mschreiber16** [reply]
> Unfortunately I still can't replicate this making it difficult to figure out where/how to fix. Which approach did you use to install Gauge itself?

Gauge was installed using installer from Gauge website and any updates to gauge would have been performed in the command line

**Reply by mschreiber16** [reply]
> Do you happen to use/develop for golang yourselves on these hosts? Do you have `GOARCH` or `GOHOSTARCH` set in environment or anything like that on these machines?
> 
> I can only find things like [golang/go#64530](https://github.com/golang/go/issues/64530)
> 
> I've tried rebuilding the plugin in https://github.com/getgauge/spectacle/releases/tag/v0.2.1 in a desperate attempt to see if that does anything to fix the issue. Can you let me know if `0.2.1` fixes the problem?

We do not use golang or goarch or gohostarch.   0.2.1 has the same issue as originally stated.

**Reply by mschreiber16** [reply]
> Yeah should be.
> 
> At least that helps confirm that something regressed (likely with the golang upgrade) in the 0.2.0 version of the plugin 😓

Tried 0.1.4 on Windows 10 and that failed with the following. 
[Gauge]
Failed to generate docs. rpc error: code = ResourceExhausted desc = grpc: received message larger than max (10566014 vs. 10485760)
Guessing the entire project was too large. I reran Spectacle using a specific folder with less specs and that seemed to work on Windows 10.

**Reply by chadlwilson** [reply]
I'm not sure in any case, as I can't replicate any such issue; while following the same architecture and versions you describe. 

There's clearly some edge case, or some combination of OS configuration/software that can trigger the main issue here (or some extra windows package whose installation makes the problem go away), so I don't think we're going to get anywhere unless someone tries to figure it out themselves, or provides a way to recreate the issue on a "clean" neutral environment/virtual machine outside whatever corporate/enterprise configuration they are using.
